perception/autoware_tensorrt_bevdet/src/bevpool_plugin.cu

```diff
@@ -0,0 +1,364 @@
+          channel, n_intervals, map_size, img_size, reinterpret_cast<const __half *>(inputs[0]),
+          reinterpret_cast<const __half *>(inputs[1]), reinterpret_cast<const int *>(inputs[2]),
+          reinterpret_cast<const int *>(inputs[3]), reinterpret_cast<const int *>(inputs[4]),
+          reinterpret_cast<const int *>(inputs[5]), reinterpret_cast<const int *>(inputs[6]),
+          reinterpret_cast<float *>(outputs[0]));
+      }
+      break;
+    case int(DataType::kHALF):
+      if (inputDesc[0].type == DataType::kFLOAT) {
+        // fp32 fp16
+        bev_pool_v2_kernel<float, __half><<<grid, block, 0, stream>>>(
+          channel, n_intervals, map_size, img_size, reinterpret_cast<const float *>(inputs[0]),
+          reinterpret_cast<const float *>(inputs[1]), reinterpret_cast<const int *>(inputs[2]),
+          reinterpret_cast<const int *>(inputs[3]), reinterpret_cast<const int *>(inputs[4]),
+          reinterpret_cast<const int *>(inputs[5]), reinterpret_cast<const int *>(inputs[6]),
+          reinterpret_cast<__half *>(outputs[0]));
+      } else {
+        // fp16 fp16
+        bev_pool_v2_kernel<__half, __half><<<grid, block, 0, stream>>>(
+          channel, n_intervals, map_size, img_size, reinterpret_cast<const __half *>(inputs[0]),
+          reinterpret_cast<const __half *>(inputs[1]), reinterpret_cast<const int *>(inputs[2]),
+          reinterpret_cast<const int *>(inputs[3]), reinterpret_cast<const int *>(inputs[4]),
+          reinterpret_cast<const int *>(inputs[5]), reinterpret_cast<const int *>(inputs[6]),
+          reinterpret_cast<__half *>(outputs[0]));
+      }
+      break;
+    default:  // should NOT be here
+      RCLCPP_ERROR(rclcpp::get_logger("BEVPoolPlugin"), "\tUnsupport datatype!");
+  }
+  return 0;
+}
+
+void BEVPoolPlugin::destroy() noexcept
+{
+  delete this;
+  return;
+}
+
+int32_t BEVPoolPlugin::initialize() noexcept
+{
+  return 0;
+}
+
+void BEVPoolPlugin::terminate() noexcept
+{
+  return;
+}
+
+size_t BEVPoolPlugin::getSerializationSize() const noexcept
+{
+  return sizeof(m_);
+}
+
+void BEVPoolPlugin::serialize(void * buffer) const noexcept
+{
+  memcpy(buffer, &m_, sizeof(m_));
+  return;
+}
+
+void BEVPoolPlugin::setPluginNamespace(const char * pluginNamespace) noexcept
+{
+  namespace_ = pluginNamespace;
+  return;
+}
+
+const char * BEVPoolPlugin::getPluginNamespace() const noexcept
+{
+  return namespace_.c_str();
+}
+
+const char * BEVPoolPlugin::getPluginType() const noexcept
+{
+  return BEVPOOL_PLUGIN_NAME;
+}
+
+const char * BEVPoolPlugin::getPluginVersion() const noexcept
+{
+  return BEVPOOL_PLUGIN_VERSION;
+}
+
+void BEVPoolPlugin::attachToContext(
+  cudnnContext * contextCudnn, cublasContext * contextCublas, IGpuAllocator * gpuAllocator) noexcept
+{
+  return;
+}
+
+void BEVPoolPlugin::detachFromContext() noexcept
+{
+  return;
+}
+
+// class BEVPoolPluginCreator
+PluginFieldCollection BEVPoolPluginCreator::fc_{};
```

Comment:
Similar to the `BEVPoolPlugin` class, consider wrapping the `BEVPoolPluginCreator` class within the same namespace for consistency and better organization.